scaleway/resource_compute_instance_server.go

```diff
@@ -0,0 +1,297 @@
+package scaleway
+
+import (
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/utils"
+)
+
+func resourceScalewayComputeInstanceServer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayComputeInstanceServerCreate,
+		Read:   resourceScalewayComputeInstanceServerRead,
+		Update: resourceScalewayComputeInstanceServerUpdate,
+		Delete: resourceScalewayComputeInstanceServerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The name of the server",
+			},
+			"image": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "The base image of the server", // TODO: add in doc example with UUID or distrib name
+				ValidateFunc: validationUUID(),
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The instance type of the server", // TODO: link to scaleway pricing in the doc
+			},
+			"tags": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The tags associated with the server",
+			},
+			"security_group_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The security group the server is attached to", // TODO: add this field in CreateServerRequest (proto)
+			},
+			//"root_volume": {
+			//	Type:     schema.TypeMap,
+			//	Optional: true,
+			//	ForceNew: true,
+			//	Elem: &schema.Resource{
+			//		Schema: map[string]*schema.Schema{
+			//			"size": {
+			//				Type:     schema.TypeString,
+			//				Optional: true,
+			//			},
+			//			"id": {
+			//				Type:     schema.TypeString,
+			//				Computed: true,
+			//			},
+			//		},
+			//	},
+			//	Description: "Root volume attached to the server on creation",
+			//},
+			"enable_ipv6": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
```

Comment:
default true ?

Reply:
IPv6 support is not really native. Some default to `false` is better imho.

Reply:
let's keep default false for now